pytorch_lightning/core/lightning.py

```diff
@@ -304,6 +304,13 @@ def log_dict(
                 tbptt_reduce_fx=tbptt_reduce_fx,
             )
 
+    def write_prediction(self, name, value, filename='predictions.pt'):
+        self.trainer.evaluation_loop.predictions._add_prediction(name, value, filename)
+
+    def write_prediction_dict(self, predictions_dict, filename='predictions.pt'):
+        for k, v in predictions_dict.items():
+            self.write_prediction(k, v, filename)
+
```

Comment:
does this write out as a CSV file? do you think this makes more sense as a callback?

Reply:
writes to list of dictionaries via torch.save. making predictions is different for every use case, so this way lets user define how they want to do it.

Could be achieved with callback (in fact, that's how I originally was doing this), but that would only work for 1 use case.